src/service/action-impl.js

```diff
@@ -501,27 +508,29 @@ export class ActionService {
    * @param {?ActionEventDef} event
    * @param {!ActionTrust} trust
    * @param {?JsonObject=} opt_args
+   * @return {boolean} True if the element has an action.
    * @private
    */
   action_(source, actionEventType, event, trust, opt_args) {
     const action =
         this.findAction_(source, actionEventType);
     if (!action) {
-      return;
+      return false;
     }
     // Use a pseudo-UUID to uniquely identify this sequence of actions.
     // A sequence is all actions triggered by a single event.
     const sequenceId = Math.random();
     // Invoke actions serially, where each action waits for its predecessor
     // to complete. `currentPromise` is the i'th promise in the chain.
+    /** @type {?Promise} */
     let currentPromise = null;
     action.actionInfos.forEach(({target, args, method, str}) => {
       const dereferencedArgs = dereferenceArgsVariables(args, event, opt_args);
       const invokeAction = () => {
         const node = this.getActionNode_(target);
         if (!node) {
           this.error_(`Target "${target}" not found for action [${str}].`);
-          return;
+          return false;
```

Comment:
Revert; returning this in a `forEach` is inert.

Reply:
done